Slim/Plugin/Podcast/Search.pm

```diff
@@ -0,0 +1,111 @@
+package Slim::Plugin::Podcast::Search;
+
+# Logitech Media Server Copyright 2005-2020 Logitech.
+
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License, 
+# version 2.
+
+use strict;
+
+use JSON::XS::VersionOneAndTwo;
+use XML::Simple;
+
+use Slim::Utils::Cache;
+use Slim::Utils::Log;
+use Slim::Utils::Prefs;
+use Slim::Utils::Strings qw(string cstring);
+
+my $log   = logger('plugin.podcast');
+my $prefs = preferences('plugin.podcast');
+
+my @providers = ( {
+		name   => 'Apple/iTunes',
+		result => 'results',
+		feed   => 'feedUrl',
+		title  => 'collectionName',
+		image  => ['artworkUrl600', 'artworkUrl100'],
+		setup => sub {
+			my $url = 'https://itunes.apple.com/search?media=podcast&term=' . shift;
+			my $country = $prefs->get('country');			
+			return $url .= "&country=$country" if $country;
+		},
+	}, {
+		name  => 'GPodder', 
+		title => 'title',
+		feed  => 'url',
+		image =>  ['scaled_logo_url', 'logo_url'],
+		setup => sub {
+			return 'https://gpodder.net/search.json?q=' . shift;
+		},
+	}
+);
+
+sub searchHandler {
+	my ($client, $cb, $args) = @_;
+
+	my ($provider) = grep { $_->{name} eq $prefs->get('provider') } @providers;
+	$provider ||= $providers[0];
+
+	my $request = HTTP::Request->new('GET');
+	my $url = $provider->{setup}->($args->{search}, $request);
+	my $cache = Slim::Utils::Cache->new();
+	
+	# try to get these from cache
+	if (my $items = $cache->get('podcast-search-' . $url)) {
+		$cb->( { items => $items } );
+		return;
+	}
+
+	# if not found in cache then re-acquire
+	my $http = Slim::Networking::Async::HTTP->new;		
```

Comment:
Wouldn't `SimpleAsyncHTTP` work? It allows you to pass additional headers:

```perl
$http->post("www.somewhere.net", 'Content-Type' => 'application/x-foo', 'Other-Header' => 'Other Value', 'conent goes here');
```

It would take care of some of the caching already etc.

Reply:
doine